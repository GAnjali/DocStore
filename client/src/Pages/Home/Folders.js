import React from 'react';
import folderIcon from "./../../Util/index.png"

const Folders = (props) => {
    return (
        <div className="folders-block">
            <span className={"heading"}> Folders </span>
            <hr/>
            <div className={"folders-list"}>
                {props.folders.map(folder => (
                    <div className={"folder"}>
                        <img src={folderIcon} className={"folder-icon"}/>
                        <p>{folder.name}</p>
                    </div>
                ))}
            </div>
        </div>
    )
};
export default Folders;