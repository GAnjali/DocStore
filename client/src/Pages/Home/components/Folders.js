import React from "react";
import folderIcon from "../styles/icons/foldericon.png";
import optionsIcon from "../styles/icons/optionsIcon.jpg";
import {DELETE} from "../../../AppConstants";

const Folders = props => {
  return (
    <div className="folders-block">
      <span className={"heading"}> Folders </span>
      <hr />
      <div className={"folders-list"}>
        {props.folders.map((folder, i) => (
          <div className={"folder"} id={i} key={i} onClick={props.handleFolderClick}>
            <img src={folderIcon} className={"folder-icon"} id={i} />
            <p id={i}>{folder.name}</p>
            <div className={"folderoptions"} id={i}>
              <img src={optionsIcon} className={"folderoptions-icon"} id={i} />
              <div className="fileoptions-content" id={i}>
                <button className={"folder-delete"} id={i}>
                    {DELETE}
                </button>
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};
export default Folders;