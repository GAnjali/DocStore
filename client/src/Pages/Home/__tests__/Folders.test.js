import React from "react";
import {shallow} from "enzyme";
import Folders from "../Folders";

describe("Folders component tests", () => {
    let folders;
    beforeEach(() => {
        folders = [];
    });

    it("Should render without crashing", () => {
        shallow(<Folders folders={folders}/>);
    });

    it('should match the Sidebar snapshot', function () {
        const folderComponent = shallow(<Folders folders={folders}/>);
        expect(folderComponent).toMatchSnapshot();
    });
});
