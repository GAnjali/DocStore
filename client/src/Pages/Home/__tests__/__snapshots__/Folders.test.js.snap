// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folders component tests should match the Sidebar snapshot 1`] = `ShallowWrapper {}`;
